File: doc/codes/python/examples/api/command/lowlevel/mod-bank.py
from autd3.commands import ChangeModulationBank, ConfigModulation, WriteModulationBuffer
from autd3.units import Hz
from autd3.value import LoopBehavior, ModulationBank, SamplingConfig, TransitionMode
from autd3_modulation import SineOption, modulation_buffer, sine

bank = ModulationBank.B0
offset = 0
buffer = modulation_buffer()
sine(150.0 * Hz, SineOption(), buffer)
data = buffer
# ANCHOR: write
WriteModulationBuffer(bank, offset, data)
# ANCHOR_END: write
config = SamplingConfig.FREQ_4K
size = len(data)
loop_behavior = LoopBehavior.Infinite
# ANCHOR: config
ConfigModulation(
    bank,
    config,
    size,
    loop_behavior,
)
# ANCHOR_END: config
transition_mode = TransitionMode.Immediate
# ANCHOR: change
ChangeModulationBank(
    bank,
    transition_mode,
)
# ANCHOR_END: change

File: doc/codes/rust/examples/api/command/lowlevel/mod-bank.rs
use anyhow::Result;

use autd3_rs::commands::{ChangeModulationBank, ConfigModulation, WriteModulationBuffer};
use autd3_rs::units::Hz;
use autd3_rs::value::{LoopBehavior, ModulationBank, SamplingConfig, TransitionMode};
use autd3_rs_modulation::{SineOption, modulation_buffer, sine};

#[tokio::main(flavor = "multi_thread")]
async fn main() -> Result<()> {
    let bank = ModulationBank::B0;
    let offset = 0;
    let mut buffer = modulation_buffer();
    sine(150 * Hz, &SineOption::default(), &mut buffer)?;
    let data = &buffer;
    // ANCHOR: write
    WriteModulationBuffer { bank, offset, data };
    // ANCHOR_END: write
    let config = SamplingConfig::FREQ_4K;
    let size = data.len();
    let loop_behavior = LoopBehavior::Infinite;
    // ANCHOR: config
    ConfigModulation {
        bank,
        config,
        size,
        loop_behavior,
    };
    // ANCHOR_END: config
    let transition_mode = TransitionMode::Immediate;
    // ANCHOR: change
    ChangeModulationBank {
        bank,
        transition_mode,
    };
    // ANCHOR_END: change
    Ok(())
}
